File: src/main/java/ca/dijital/hydro/Station.java
package ca.dijital.hydro;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Calendar;
import java.util.Date;
import java.util.TimeZone;

import org.postgis.PGgeometry;
import org.postgis.Point;

public class Station {

	private long gid;
	private String id;
	private String name;
	private String prov;
	private double lat;
	private double lon;
	private Date lastUpdate;
	private static Calendar cal = Calendar.getInstance();
	
	public static Station fromResultSet(ResultSet rslt) throws SQLException {
		Station s = new Station();
		Point pt = (Point) ((PGgeometry) rslt.getObject("geom")).getGeometry();
		s.setGid(rslt.getLong("gid"));
		s.setId(rslt.getString("id"));
		s.setLat(pt.y);
		s.setLon(pt.x);
		s.setName(rslt.getString("name"));
		s.setProv(rslt.getString("prov"));
		Date dt = rslt.getTimestamp("lastupdate");
		if(dt != null) {
			cal.setTime(dt);
			cal.setTimeZone(TimeZone.getTimeZone("UTC"));
			s.setLastUpdate(cal.getTime());
		}
		return s;
	}
	
	public long getGid() {
		return gid;
	}
	public void setGid(long gid) {
		this.gid = gid;
	}
	public String getId() {
		return id;
	}
	public void setId(String id) {
		this.id = id;
	}
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	public String getProv() {
		return prov;
	}
	public void setProv(String prov) {
		this.prov = prov;
	}
	public double getLat() {
		return lat;
	}
	public void setLat(double lat) {
		this.lat = lat;
	}
	public double getLon() {
		return lon;
	}
	public void setLon(double lon) {
		this.lon = lon;
	}

	public Date getLastUpdate() {
		return lastUpdate;
	}

	public void setLastUpdate(Date lastUpdate) {
		this.lastUpdate = lastUpdate;
	}
	
}

File: svc/station.js
class Station {


	constructor(params) {
		this.sid = 0;
		this.id = null;
		this.name = null;
		this.prov = null;
		this.lat = Number.NaN;
		this.lon = Number.NaN;
		this.lastupdate = null;
		if(params) {
			for(let [k, v] of Object.entries(params))
				this[k] = v;
		}	
	}	

	getGid() {
		return this.gid;
	}
	setGid(gid) {
		this.gid = gid;
	}
	getId() {
		return this.id;
	}
	setId(id) {
		this.id = id;
	}
	getName() {
		return this.name;
	}
	setName(name) {
		this.name = name;
	}
	getProv() {
		return this.prov;
	}
	setProv(prov) {
		this.prov = prov;
	}
	getLat() {
		return this.lat;
	}
	setLat(lat) {
		this.lat = lat;
	}
	getLon() {
		return this.lon;
	}
	setLon(lon) {
		this.lon = lon;
	}

	getLastUpdate() {
		return this.lastupdate;
	}

	setLastUpdate(lastupdate) {
		this.lastupdate = lastupdate;
	}
	
}

module.exports = Station;